### ISSUE TITLE
Review FreeCEN2 analytics

### ISSUE THREAD

**Issue by richpomfret** [FreeUKGen/FreeCENMigration#494]
Since the gdpr cookie change, we've seen a daily spike in users. This may be as a result of the changes we made rather than actual users (but hopefully not!). The same applies on FR.

**Comment by smrr723** [comment]
Have found what appears to be the cause here, after looking into the IP Anonymisation issue.

The line `ga('send', 'pageview');` is wrongly placed, and as a result, it's being fired 4 times when a user visits and accepts the cookie policy.  It should only be getting fired twice - once for the first page visit, and again on the page reload after they accept the cookie policy.  This explains the 4 x spike in pageviews following the cookie code changes.

```      window.update_third_party_cookies_user_preference = function (preference) {
        if(preference == 'accept') {
          ga('create', '<%= google_analytics_tracking_id %>', 'auto');
        } else if(preference == 'deny') {
          delete_cookie('_ga');
          delete_cookie('_gid');
          delete_cookie('_gat');
          ga('create', '<%= google_analytics_tracking_id %>', {'storage': 'none', 'cookieExpires': 0 });
        };
        ga('send', 'pageview');```

Currently testing a fix for this.

**Comment by smrr723** [comment]
As discussed earlier, have a solution for this which uses and depends on implementing #475 to make things simpler.  Have basically had to scrap the existing Analytics cookie code in the end and just start over. 

Will try to get the same solution on FR this sprint also.

**Comment by richpomfret** [comment]
I am unsure just how this might be adjusted or changed when considering Tag Manager - however, we DO want to trigger the anonymised IP for a user when they first visit the site - we also ensure any third party cookies (such as GA advertisement/marketing features) are deactivated with a flag set to 'FALSE' until the user explicitly consents - we can then fire it off just fine. Complexity lies in how we deal with user cases where they go into the cookie toggle page and switch advertising (or tag manager cookies?) on or off.

**Comment by smrr723** [comment]
If we do want to trigger the Anon IP for all new users then that makes things easier. I've gone ahead an set up a Tag Manager account for FC2 with `anonIP: true` by default.  This will anonymize their IP stored in the default analytics cookies that gets created on the first visit.

I've removed the Analytics code and replaced it with Tag Manager code - so I'll make another production update shortly to trial it for a few days.  

Regarding the advertising stuff - all ad cookies are disabled by default at the moment.  The analytics cookies we set, `ga, gat, gid` store no ad info.  There's an analytics cookie `gac` which relates to advertising conversion tracking but we don't have that being set by our analytics code.

Also, the existing code is still able to handle deletion of the analytics cookies if users toggle off so the last part shouldn't be an issue.  There aren't any cookies for GTM, it's just a container for our analytics code really.
